qutebrowser/config/config.py

```diff
@@ -504,13 +503,19 @@ class StyleSheetObserver(QObject):
         _stylesheet: The stylesheet template to use.
     """
 
-    def __init__(self, obj, stylesheet):
-        super().__init__(parent=obj)
+    def __init__(self, obj, stylesheet, update):
+        super().__init__()
         self._obj = obj
+        self.update = update
+
+        # We only need to hang around if we are asked to update.
+        if self.update:
+            self.setParent(self._obj)
         if stylesheet is None:
             self._stylesheet = obj.STYLESHEET
         else:
             self._stylesheet = stylesheet
+        self._register()
```

Comment:
I'd rather keep this as a separate method. Updating the stylesheet and all just because we created a `StyleSheetObserver` seems like a weird API to me.